### PR TITLE
[RSM] EvidenceSubmittedList component for Dispute Outcome View

### PR DESCRIPTION
**Pull request by mgascam**
Fixes [RSM-1165](https://linear.app/a8c/issue/RSM-1165/evidencesubmittedlist-presentational-component)

## Summary

Adds the `EvidenceSubmittedList` presentational component. It takes `EvidenceFieldStatus[]` (the output of `getExpectedFieldStatus(reason, productType, evidence)` from Slice 2) and renders a tri-state list:

- `provided`: green check, default label color
- `expected_missing`: red close icon, red label, visible "— Not provided" suffix
- `optional_missing`: grey dash, grey label, visible "— Not provided" suffix

Provided rows announce "Provided" via a visually-hidden span so screen readers hear a state for every item, even though the green check stands alone visually.

## Scope

This PR is intentionally small and presentational. It only touches the EvidenceSubmittedList component.

Screenshot 2026-05-07 at 19 13 23

Screenshot 2026-05-07 at 19 13 35

Screenshot 2026-05-07 at 20 22 51

## Fixture approach

Reference fixtures live in `client/payment-details/dispute-outcome/__fixtures__/evidence-statuses.ts`:

- `wonFraudulentPhysical`: typical clean win (5 provided, 2 optional missing)
- `lostProductUnacceptablePhysical`: typical loss (1 provided, 3 expected missing, 2 optional missing)

These mirror the shape that `getExpectedFieldStatus` returns at runtime, but are hand-built so tests and visual review can exercise the tri-state surface without seeding a full dispute. Two tests in `__tests__/evidence-submitted-list.test.tsx` consume these fixtures directly to assert the expected state mix per outcome.

## Browser testing (optional)

> [!NOTE]
>  The component is not visible in the browser by default. Testing is optional for now; verifying with the browser will be much easier when we wire the component with the dispute data. To see it rendered against the existing test disputes, apply this temporary diff locally:

**Step 1.** In `client/payment-details/dispute-outcome/index.tsx`, replace the placeholder body so the component renders one of the fixtures based on dispute status:

```tsx
/** @format **/

import React from 'react';
import type { ChargeDispute } from 'wcpay/types/charges';
import EvidenceSubmittedList from './evidence-submitted-list';
import {
    wonFraudulentPhysical,
    lostProductUnacceptablePhysical,
} from './__fixtures__/evidence-statuses';

interface DisputeOutcomeViewProps {
    dispute: ChargeDispute;
}

const DisputeOutcomeView: React.FC< DisputeOutcomeViewProps > = ( { dispute } ) => {
    const fixture =
        dispute.status === 'won'
            ? wonFraudulentPhysical
            : lostProductUnacceptablePhysical;
    return (
        <div style={ { padding: '24px' } }>
            <h3 style={ { margin: '0 0 16px' } }>Evidence Submitted</h3>
            <EvidenceSubmittedList fields={ fixture } />
        </div>
    );
};

export default DisputeOutcomeView;
```

**Step 2.** In `client/payment-details/summary/index.tsx`, inside `renderDisputeDetails(...)` (around line 104), prepend an early return so the component renders for any dispute regardless of feature flag or status:

```tsx
if ( dispute ) {
    return <DisputeOutcomeView dispute={ dispute } />;
}
```

**Step 3.** Build and view:

```bash
npm run build:client
# Open any dispute on /payments/disputes
```

**Step 4.** Revert when done:

```bash
git checkout client/payment-details/dispute-outcome/index.tsx \
              client/payment-details/summary/index.tsx
```

## Test plan

- Tests pass
- Eyeball the screenshots above against the mocks 17pWAxW0l_5mSUNzSqbZgpi6sRxJOwCKtZdGrdOsWofg-gdoc
- (Optional) Apply the Browser testing diff and visit a dispute page

## Out of scope

- Real data wiring (upcoming)
- Tracks events (deferred to its own slice)
- "What Could Help" links next to missing fields (separate component)

### CODE REVIEW

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+							{ label }
+						</span>
+						{ state === 'provided' ? (
+							<VisuallyHidden>
+								{ renderStateLabel( state ) }
+							</VisuallyHidden>
+						) : (
+							<span className="dispute-outcome-evidence-list__state">
+								{ renderStateLabel( state ) }
+							</span>
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/evidence-submitted-list.tsx:92]: Good catch. Switched from CSS ::before to real DOM content: an aria-hidden="true" span around the visible em-dash plus an explicit { ' ' } text node between the label and state span. SRs now hear a word boundary; sighted users still see "— Not provided".

File: client/payment-details/dispute-outcome/__fixtures__/evidence-statuses.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/** @format */
+
+/**
+ * Hand-built `EvidenceFieldStatus[]` fixtures for the EvidenceSubmittedList
+ * component. These mirror the shape returned by `getExpectedFieldStatus`
+ * (see `client/disputes/new-evidence/helpers.ts`) so tests can exercise
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/__fixtures__/evidence-statuses.ts]: Fixed. Updated to evidence-field-status.ts.

File: client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+				fields={ [
+					provided( 'A' ),
+					{ ...expectedMissing( 'B' ), key: 'refund_policy' },
+				] }
+			/>
+		);
+
+		expect( consoleError ).not.toHaveBeenCalledWith(
+			expect.stringMatching( /unique "key" prop/ )
+		);
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx]: Agreed, the original matcher missed multi-arg calls. Replaced with a mock.calls.some(...) scan that flattens args and checks the substring.

File: client/payment-details/dispute-outcome/style.scss [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+	&__text {
+		flex: 1 1 auto;
+		min-width: 0; // allow children to shrink below their intrinsic width and wrap
+	}
+
+	&__label {
+		// inline so it flows with __state in the same wrapping text block
+	}
+
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/style.scss]: Fixed

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+};
+
+const renderStateLabel = ( state: EvidenceFieldState ): string => {
+	switch ( state ) {
+		case 'provided':
+			return __( 'Provided', 'woocommerce-payments' );
+		case 'expected_missing':
+		case 'optional_missing':
+			return __( 'Not provided', 'woocommerce-payments' );
+	}
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/evidence-submitted-list.tsx]: Right, neither shape nor color is available to screen reader users. Added a VisuallyHidden severity hint inside the state span: "Not provided (required)" or "Not provided (optional)". Visible copy stays unified per the mock; SRs now distinguish all three states.

File: client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+		// the warning. Inspect every call's args and look for the substring.
+		const sawUniqueKeyWarning = consoleError.mock.calls.some( ( args ) =>
+			args.some(
+				( arg ) =>
+					typeof arg === 'string' && /unique "key" prop/.test( arg )
+			)
+		);
+		expect( sawUniqueKeyWarning ).toBe( false );
+		consoleError.mockRestore();
+	} );
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx]: Renamed the existing test to reflect what it actually checks ("does not emit a React missing-key warning when fields have valid keys") and added a separate, more direct test that intentionally uses duplicate
field.key values and asserts React's duplicate-key warning fires. That directly validates the contract that field.key becomes the React key — array index or any other keying strategy would mask the collision and the new test would fail.

File: client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+		// so single-arg matchers like toHaveBeenCalledWith( stringMatching ) miss
+		// the warning. Inspect every call's args and look for the substring.
+		const sawUniqueKeyWarning = consoleError.mock.calls.some( ( args ) =>
+			args.some(
+				( arg ) =>
+					typeof arg === 'string' && /unique "key" prop/.test( arg )
+			)
+		);
+		expect( sawUniqueKeyWarning ).toBe( false );
+		consoleError.mockRestore();
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx]: Right, explicit mockRestore() at the end of the test only runs on the happy path. Replaced both with an afterEach( jest.restoreAllMocks ) at the describe level — runs unconditionally regardless of whether the test passed or threw, so spy state never bleeds across tests.

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+							<VisuallyHidden>
+								{ renderStateLabel( state ) }
+							</VisuallyHidden>
+						) : (
+							<span className="dispute-outcome-evidence-list__state">
+								<span aria-hidden="true">{ '— ' }</span>
+								{ renderStateLabel( state ) }
+								<VisuallyHidden>
+									{ ` (${ renderSeverityHint( state ) })` }
+								</VisuallyHidden>
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/evidence-submitted-list.tsx:91]: Right — single-word __() strips translator context, and template-literal parens hard-code English bracket conventions. Refactored to use sprintf for the wrapper (with a translator comment) and _x for the severity word with explicit context: "severity of a missing evidence field, read by screen readers". Locales can now adapt punctuation and translators have the context they need.

[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/evidence-submitted-list.tsx:91]: Took another pass at this and simplified further. The sprintf wrapping translated " (%s)" as its own .po entry, but that's just punctuation and a placeholder with no semantic content for translators to act on. For a screen-reader-only suffix the theoretical benefit (locales adapting bracket conventions) doesn't outweigh the noise of a near-empty translation entry. renderSeverityHint now returns just the translated severity word; the leading space for the SR word boundary moves to the JSX call site. Translators see only the two meaningful words ("required" / "optional") plus their context comments.

File: client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+		render(
+			<EvidenceSubmittedList
+				fields={ [ expectedMissing( 'A' ), optionalMissing( 'B' ) ] }
+			/>
+		);
+
+		const stateNodes = screen.getAllByText( /^not provided$/i );
+		expect( stateNodes ).toHaveLength( 2 );
+		stateNodes.forEach( ( node ) => {
+			expect( node.className ).toMatch(
```
[the remark this answers is not in the document]
Comment (mgascam) [reply client/payment-details/dispute-outcome/__tests__/evidence-submitted-list.test.tsx:135]: The test passes because RTL's getByText matches an element's direct text children, not its full textContent. The em-dash and severity hint live in child elements (<span aria-hidden> and <VisuallyHidden>), so the __state span's own direct text is exactly "Not provided" — which the anchored regex matches. The test correctly returns the two __state elements without false positives. Leaving as-is; the precise matcher is intentional and would catch typos like "Not provided yet" that a non-anchored regex would miss.

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+};
+
+const renderStateIcon = ( state: EvidenceFieldState ): JSX.Element => {
+	switch ( state ) {
+		case 'provided':
+			return <Icon icon={ check } />;
+		case 'expected_missing':
+			return <Icon icon={ closeSmall } />;
+		case 'optional_missing':
+			return <>—</>;
```
Comment (dmvrtx) [inline client/payment-details/dispute-outcome/evidence-submitted-list.tsx]: Can we use a suitable `Icon` here as well? It seems to create inconsistency in the response.

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+	optional_missing: 'optional-missing',
+};
+
+const renderStateLabel = ( state: EvidenceFieldState ): string => {
+	switch ( state ) {
+		case 'provided':
+			return __( 'Provided', 'woocommerce-payments' );
+		case 'expected_missing':
+		case 'optional_missing':
+			return __( 'Not provided', 'woocommerce-payments' );
```
Comment (dmvrtx) [inline client/payment-details/dispute-outcome/evidence-submitted-list.tsx]: Can we use `stateModifiers` constants here instead of strings?

**Review by dmvrtx — APPROVED** [review]
Looks good! There are a couple of nitpicks, but nothing blocking.

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx (hunk 10 above)
Comment (mgascam) [reply client/payment-details/dispute-outcome/evidence-submitted-list.tsx]: Swapped for `lineSolid` which is the closest neutral horizontal-line glyph in @wordpress/icons. 

icon-preview

[b7245ba](https://github.com/Automattic/woocommerce-payments/pull/11677/commits/b7245ba41c32526b08bb2e84bf7052e581227de8) Thanks!

File: client/payment-details/dispute-outcome/evidence-submitted-list.tsx (hunk 11 above)
Comment (mgascam) [reply client/payment-details/dispute-outcome/evidence-submitted-list.tsx]: Done [b7245ba](https://github.com/Automattic/woocommerce-payments/pull/11677/commits/b7245ba41c32526b08bb2e84bf7052e581227de8) thanks!
